test/Implementation/ModelTraitImplementation.php

```diff
@@ -134,9 +134,9 @@ public function setCollectionProp(array $collectionProp)
     }
 
     /**
-     * @param ModelTraitImplementation $collectionProp
+     * @param ModelTraitImplementation|array $collectionProp
      */
-    public function addCollectionProp(ModelTraitImplementation $collectionProp)
+    public function addCollectionProp(ModelTraitImplementation|array $collectionProp)
```

Comment:
No, this can't be an array. Why have you changed this?

Reply:
Data set 5:
`[
                    "collectionProp" => [
                        $prop1Data,
                        $prop2Data,
                        $prop3Data
                    ]
]`
is an array when it's passed to `addCollectionProp` in the loadData function

Reply:
In setCollectionProp, the elements of the input array are also arrays. I could convert them to instances of ModelTraitImplementations before calling addCollectionProp

Reply:
it shouldn't be, it should loop over that array and add each entry individually

Reply:
That conversion is meant to happen automatically, thats one of the main points of clay

Reply:
I think this is failing because t's not detecting the class type due to the issue in the other open comment. 

You can definitely revert these "array" changes; the tests weren't in error, the code is.

Reply:
Will do